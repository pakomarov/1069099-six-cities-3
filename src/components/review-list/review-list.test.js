import React from 'react';
import renderer from 'react-test-renderer';
import ReviewList from './review-list.jsx';


const reviews = [
  {
    id: 1,
    text: `Very nice place`,
    rating: 5,
    user: {
      name: `Ivan`,
      avatar: `img/avatar-angelina.jpg`,
    },
    date: new Date(`2019-05-08T14:13:56.569Z`),
  },
  {
    id: 2,
    text: `Bad bad bad`,
    rating: 1,
    user: {
      name: `John Doe`,
      avatar: `img/avatar-angelina.jpg`,
    },
    date: new Date(`2016-04-06T14:13:56.569Z`),
  },
  {
    id: 3,
    text: `Good enough`,
    rating: 4,
    user: {
      name: `Marina`,
      avatar: `img/avatar-angelina.jpg`,
    },
    date: new Date(`1917-06-26T14:13:56.569Z`),
  },
];


it(`Should match snapshot of ReviewList`, () => {
  const tree = renderer
    .create(<ReviewList
      reviews={reviews}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
