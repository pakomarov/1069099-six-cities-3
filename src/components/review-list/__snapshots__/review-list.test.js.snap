// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot of ReviewList 1`] = `
Array [
  <h2
    className="reviews__title"
  >
    Reviews Â· 
    <span
      className="reviews__amount"
    >
      3
    </span>
  </h2>,
  <ul
    className="reviews__list"
  >
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="img/avatar-angelina.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Ivan
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "100%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          Very nice place
        </p>
        <time
          className="reviews__time"
          dateTime="2019-05-08"
        >
          May 2019
        </time>
      </div>
    </li>
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="img/avatar-angelina.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          John Doe
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "20%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          Bad bad bad
        </p>
        <time
          className="reviews__time"
          dateTime="2016-04-06"
        >
          April 2016
        </time>
      </div>
    </li>
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="img/avatar-angelina.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Marina
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "80%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          Good enough
        </p>
        <time
          className="reviews__time"
          dateTime="1917-06-26"
        >
          June 1917
        </time>
      </div>
    </li>
  </ul>,
]
`;
